### PR TITLE
Add example blink project for nucleo-f401re board

### PR DESCRIPTION
**Pull request by hyves42**
This is the same code as the nucleo-f411re board. I just modified the linker configuration to support the f401re board.

### CODE REVIEW

**Comment by malvira** [comment]
:+1: saved me a lot of time trying to figure this out. Thanks! :smiley:

File: examples/stm32/f4/nucleo-f401re/nucleo-f401re.ld [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Linker script for Nucleo F411RE  (STM32F411RE, 512K flash, 128K RAM). */
```
Comment (karlp) [inline examples/stm32/f4/nucleo-f401re/nucleo-f401re.ld]: wrong comment here

File: examples/stm32/f4/nucleo-f401re/nucleo-f401re.ld [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2009 Uwe Hermann <uwe@hermann-uwe.de>
+ * Copyright (C) 2011 Stephen Caudle <scaudle@doceme.com>
```
Comment (karlp) [inline examples/stm32/f4/nucleo-f401re/nucleo-f401re.ld]: There's nothing here written by either of these people of any relevance.  Either omit it, or put your own name here for 2015/2016

File: examples/stm32/f4/nucleo-f401re/blink/blink.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2009 Uwe Hermann <uwe@hermann-uwe.de>
+ * Copyright (C) 2011 Stephen Caudle <scaudle@doceme.com>
+ * Copyright (c) 2015 Chuck McManis <cmcmanis@mcmanis.com>
```
Comment (karlp) [inline examples/stm32/f4/nucleo-f401re/blink/blink.c]: I'm unconvinced that all three of these are relevant.

File: examples/stm32/f4/nucleo-f401re/blink/Makefile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+##
+## This file is part of the libopencm3 project.
+##
+## Copyright (C) 2009 Uwe Hermann <uwe@hermann-uwe.de>
```
Comment (karlp) [inline examples/stm32/f4/nucleo-f401re/blink/Makefile]: nope.  Not Uwe, and not 2009.

**Comment by karlp** [comment]
this is covered by the miniblink repo now thanks, see https://github.com/libopencm3/libopencm3-miniblink/blob/master/boards.stm32.mk#L106

**Pull request closed without merging**